contracts/src/registry/ETHRegistrar.sol

```diff
@@ -22,16 +27,17 @@ contract ETHRegistrar is IETHRegistrar, AccessControl {
     error NameNotAvailable(string name);
     error InsufficientValue(uint256 required, uint256 provided);
 
-    IETHRegistry public immutable registry;
+    IPermissionedRegistry public immutable registry;
```

Comment:
We should probably define an interface specific to a registry-with-register-and-renew functions that's more general than IPermissionedRegistry.

Reply:
I need more guidance here. Should it have relinquish as well? what about the getters to fetch expiry, etc? and that setters for subregistry and resolver value?

Reply:
Ideally it should be a useful interface that's a superset of `IRegistry`, but not specific to `PermissionedRegistry`. So it would need to include `register`, `renew` and any other methods this contract uses, but needn't include methods this contract doesn't use unless they're naturally part of the same interface. So the getters for expiry etc would naturally be included because their setters are, for example.

Reply:
Ok, I've just updated `IRegistry` directly